async-usercalls/src/duplicated.rs

```diff
@@ -0,0 +1,168 @@
+//! this file contains code duplicated from libstd's sys/sgx
```

Comment:
Can you think of a logical division of the code in this PR between `std` and an external crate that reduces duplication?

Reply:
I think these parts can live in `std`:
- Most of the allocator (except `io_bufs.rs`)
- `callback.rs` which is the main user of the duplicated code